src/shogun/kernel/TranslationalInvariantKernel.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (w) 2016 Soumyajit De
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#include <shogun/lib/config.h>
+
+#ifndef TRANSLATIONAL_INVARIANT_KERNEL_H_
+#define TRANSLATIONAL_INVARIANT_KERNEL_H_
+
+#include <shogun/kernel/Kernel.h>
+#include <shogun/distance/Distance.h>
+
+namespace shogun
+{
+
+/** @brief The translational invariant kernel */
+class CTranslationalInvariantKernel: public CKernel
+{
+public:
+	/** Default constructor.  */
+	CTranslationalInvariantKernel();
+
+	/**
+	 * Constructor with a specified cache size.
+	 *
+	 * @param cache cache size
+	 */
+	CTranslationalInvariantKernel(int32_t cache);
+
+	/**
+	 * Constructor that initializes the kernel with two feature instances.
+	 *
+	 * @param l features of left-hand side
+	 * @param r features of right-hand side
+	 * @param cache cache size
+	 */
+	CTranslationalInvariantKernel(CFeatures *l, CFeatures *r, int32_t cache);
```

Comment:
yeah that's better. This is just to imitate the API we have for CKernel. I'll change it.